solidity/contracts/WALTHEA.sol

```diff
@@ -13,72 +13,41 @@
 // You should have received a copy of the GNU General Public License
 // along with this program.  If not, see <http://www.gnu.org/licenses/>.
 
-pragma solidity >=0.4.22;
-
-/// This contract is a copy of the WETH9 contract found at https://github.com/gnosis/canonical-weth/blob/master/contracts/WETH9.sol
-/// with the only notable changes being a solidity compiler version relaxation and minimal syntactic changes to ensure
-/// the contract compiles and produces no warnings.
-contract WETH9 {
-    string public name     = "Wrapped Ether";
-    string public symbol   = "WETH";
-    uint8  public decimals = 18;
-
-    event  Approval(address indexed src, address indexed guy, uint wad);
-    event  Transfer(address indexed src, address indexed dst, uint wad);
+pragma solidity 0.8.28;
+
+import {ERC20Permit} from "@openzeppelin/contracts/token/ERC20/extensions/ERC20Permit.sol";
+
+/// This contract is based on the WETH9 contract found at https://github.com/gnosis/canonical-weth/blob/master/contracts/WETH9.sol
+/// with the following changes:
+///    name and symbol,
+///    solidity compiler version updated to 0.8.28,
+///    reimplementation of core ERC20 functions via openzeppelin's contracts
+///    addition of the permit() function via openzeppelin's ERC20Permit contract
+contract WETH9 is ERC20Permit {
+    constructor() ERC20("Wrapped Althea", "WALTHEA") ERC20Permit("WALTHEA") {}
+
     event  Deposit(address indexed dst, uint wad);
     event  Withdrawal(address indexed src, uint wad);
 
-    mapping (address => uint)                       public  balanceOf;
-    mapping (address => mapping (address => uint))  public  allowance;
-
     fallback() external payable {
         deposit();
     }
+
     receive() external payable {
         deposit();
     }
+
     function deposit() public payable {
-        balanceOf[msg.sender] += msg.value;
+        address sender = _msgSender();
```

Comment:
Rather than updating the mapping values directly, openzeppelin exposes _mint to create tokens - this updates totalSupply as well.